File: feature-dashboard/src/main/java/fr/o80/sample/featuredashboard/DashboardService.java
package fr.o80.sample.featuredashboard;

import javax.inject.Inject;

import fr.o80.sample.lib.core.LibConfiguration;

/**
 * @author Olivier Perez
 */
public class DashboardService {

    private final LibConfiguration configuration;

    @Inject
    public DashboardService(LibConfiguration configuration) {
        this.configuration = configuration;
    }

    public String getTitle() {
        return "Dashboard title " + configuration;
    }
}

File: feature-dashboard/src/main/java/fr/o80/sample/featuredashboard/DashboardService.kt
package fr.o80.sample.featuredashboard

import javax.inject.Inject

import fr.o80.sample.lib.core.LibConfiguration

/**
 * @author Olivier Perez
 */
class DashboardService @Inject constructor(private val configuration: LibConfiguration) {

    val title: String
        get() = "Dashboard title " + configuration
}
